### PR TITLE
Issue #3511 fixed

### PR DESCRIPTION
**Pull request by snehatulsi**
Call on-toggle function of dropdown in resolve function of $animate

### CODE REVIEW

**Comment by chrisirhc** [comment]
It looks like with this change, you need to do `$browser.defer.flush()` in the dropdown.spec for the failing tests because the `toggleHandler` is now deferred.

**Comment by rvanbaalen** [comment]
Since this related issue, #3511, is labeled Angular 1.3 I've added the Angular 1.3 label here as well.

**Comment by chrisirhc** [comment]
@rvanbaalen not exactly, it's actually an ngAnimate issue but nonetheless we should try to have this fixed for next release.

**Comment by chrisirhc** [comment]
@snehatulsi , could you please add `$browser.defer.flush()` in the test code so that the tests pass?

**Comment by snehatulsi** [comment]
I don't know where is the test, and where can I add the code. Can you
please help me out.

**Comment by chrisirhc** [comment]
No problem, the tests are located at: https://github.com/angular-ui/bootstrap/blob/master/src/dropdown/test/dropdown.spec.js

You can see the failures when you run `grunt` on your command line once you've [prepared your environment](https://github.com/angular-ui/bootstrap/wiki/Development#prepare-your-environment).

**Comment by snehatulsi** [comment]
@chrisirhc I tried adding the code. I got the error $browser not defined. I tried with modifications, but failed. I don't know the test specs, I have never done testing this way. Please mention where to inject $browser dependency, and where to insert code for $browser.defer.flush()

File: src/dropdown/test/dropdown.spec.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 describe('dropdownToggle', function() {
-  var $compile, $rootScope, $document, element;
+  var $compile, $rootScope, $document, element, $browser;
 
   beforeEach(module('ui.bootstrap.dropdown'));
 
-  beforeEach(inject(function(_$compile_, _$rootScope_, _$document_) {
+  beforeEach(inject(function(_$compile_, _$rootScope_, _$document_, $browser) {
```
Comment (chrisirhc) [inline src/dropdown/test/dropdown.spec.js]: Use `_$browser_` here

**Comment by chrisirhc** [comment]
Added a comment. You need to add them just before the expect calls on the lines that they're failing.

**Comment by snehatulsi** [comment]
@chrisirhc The test passed

**Comment by karianna** [comment]
@snehatulsi Can you update this PR, it can no longer be merged against master

**Comment by wesleycho** [comment]
Can you squash your commits & rebase off of master?

**Pull request closed without merging**